src/components/tools/geometry-tool/geometry-tool.tsx

```diff
@@ -500,6 +501,7 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
     const content = this.getContent();
     const { board } = this.state;
     if (board) {
+      Logger.logTileEvent(LogEventName.TILE_UNDO, this.props.model);
```

Comment:
@sfentress Let me know if you have any thoughts on my logging approach here - I figure the idea is to pass the minimum amount of information possible to reconstruct a student's activity.

Also I went with `TILE_UNDO` instead of `UNDO` since I assume we'll have some kind of global document undo in the future.

Reply:
I would make some changes to this model, and would be happy to discuss further:

First, I think that an undo action is a different category from simply an action called "undo." In the original brainstorming document for logging, @dougmartin proposed a field called `method` in all events, which is `do` for all regular actions, and `undo` and `redo` for undo/redo actions. This allows some better filtering by researchers. The property `method: "do"` is currently in all log messages.

Also, I would make this a "tool change" event rather than a "tile change" event, because it's an action happening inside the tool.

I would propose we extract the actual undone event (so that researchers aren't forced to rely on timestamps to know what was undone) and log it fully, as if the user had done that action, but we add the `method: undo` in the log message. Something like

```
private handleUndo = () => {
    const content = this.getContent();
    const { board } = this.state;
    if (board) {
      const changeset = content.popChangeset();
      if (changeset) {
        board.showInfobox(false);
        this.setState({
          redoStack: this.state.redoStack.concat([changeset])
        });
        // unsure what to do about batched changes
        const change = safeJsonParse(changeset[0]) as JXGChange;
        // this looks like the regular event logging in geometry-content.ts
        Logger.logToolChange(LogEventName.GRAPH_TOOL_CHANGE, change.operation, change,
          content.metadata ? content.metadata.id : "",
          LogEventMethod.UNDO);                                    // <-- this is new
      }
    }

    return true;
  }
```

And in `logger.ts`:

```
  export enum LogEventMethod {
    DO,
    UNDO,
    REDO
  }

  ...

  public static logToolChange(eventName: LogEventName, operation: string, change: 
       ToolChangeEventType, toolId: string, method?: LogEventMethod) {
    if (!method) {
      method = LogEventMethod.DO;
    }
    const parameters: {[k: string]: any} = {
      toolId,
      operation,
      ...change
    };

    Logger.log(eventName, parameters, method);
  }

  // ... and pass method along to createLogMessage 
```
Thoughts?